### PR TITLE
Added documentation on Promises in TypeScript

### PR DESCRIPTION
**Pull request by felpsg**
### Description

I've created a new entry in the TypeScript section that focuses on how to use Promises in TypeScript. This entry includes examples of code and detailed explanations about its key features and methods. 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

File: content/typescript/concepts/promises/promises.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+| Title    | Description                                                                                                            | Subjects           | Tags                             | CatalogContent                   |
```
Comment (yangc95) [inline content/typescript/concepts/promises/promises.md:1]: @felpsg take a look at the documentation folder where there are templates for the front matter: https://raw.githubusercontent.com/Codecademy/docs/main/documentation/concept-entry-template.md

**Comment by yangc95** [comment]
@felpsg Nice initiative! I noticed there isn't an issue for this but no biggie. There also doesn't seem to be overlap with existing concepts so that's great. Before conducting a first review, please take a look at the template link I provided and do a quick self-review of this entry. Thanks!

**Comment by felpsg** [comment]
Hello, thank you for the feedback. However, I used a structure from an existing TypeScript file, maintaining the same format that was already there. Nonetheless, I've fixed the introduction for better readability now. I appreciate the feedback.

**Comment by yangc95** [comment]
> I used a structure from an existing TypeScript file, maintaining the same format that was already there.

@felpsg I think you may have been looking at the preview rather than the raw markdown, but not a big deal! I'll take this on for a review

File: content/typescript/concepts/promises/promises.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+```ts
+myPromise.then((value) => {
+  console.log('Promise resolved with value: ' + value);
+}).catch((error) => {
+  console.error('Promise rejected with error: ' + error);
+});
+```
+
+## `async`/`await`
```
Comment (yangc95) [inline content/typescript/concepts/promises/promises.md:55]:
```suggestion
### `async`/`await`
```

File: content/typescript/concepts/promises/promises.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+CatalogContent:
+  - 'learn-typescript'
+  - 'paths/full-stack-engineer-career-path'
+---
+
+In TypeScript, **promises** handle asynchronous operations, providing better control over the flow of code.
+
+## Creating Promises
+
+To create a new Promise, use the `new` keyword followed by `Promise`. The `Promise` constructor accepts a function which should take two parameters:
```
Comment (yangc95) [inline content/typescript/concepts/promises/promises.md:18]:
```suggestion
To create a new promise, use the `new` keyword followed by `Promise`. The `Promise` constructor accepts a function which should take two parameters:
```

**Review by yangc95 — APPROVED** [review]
@felpsg I made some direct changes. Apart from the two additional suggestions, I'd say this is ready for a second review

**Review by SSwiniarski — APPROVED** [review]
@felpsg Looks good to me.

**Comment by SSwiniarski** [comment]
@felpsg a note for future PRs, you should create a separate branch off your main fork for your edits. This allows you to do more than one PR at a time, if they're all on separate branches.

**Pull request merged**